### PR TITLE
update feed lookup to work with mercury 0.3 spec

### PR DESCRIPTION
**Pull request by FelixFan1992**

### CODE REVIEW

File: core/services/ocr2/plugins/ocr2keeper/evm21/feed_lookup.go [hunk 1]
```diff
@@ -400,11 +410,15 @@ func (r *EvmRegistry) multiFeedsRequest(ctx context.Context, ch chan<- MercuryDa
...
-	req.Header.Set("X-Authorization-Timestamp", strconv.FormatInt(ts, 10))
-	req.Header.Set("X-Authorization-Signature-SHA256", signature)
+	signature := r.generateHMAC(http.MethodGet, mercuryBatchPathV3+q.Encode(), []byte{}, r.mercury.cred.Username, r.mercury.cred.Password, ts)
+	req.Header.Set(headerContentType, applicationJson)
+	// username here is often referred to as user id
+	req.Header.Set(headerAuthorization, r.mercury.cred.Username)
+	req.Header.Set(headerTimestamp, strconv.FormatInt(ts, 10))
+	req.Header.Set(headerSignature, signature)
+	// mercury will inspect authorization headers above to make sure this user (in automation's context, this node) is eligible to access mercury
+	// and if it has an automation role. it will then look at this upkeep id to check if it has access to all the requested feeds.
```
Comment (cmalec) [inline core/services/ocr2/plugins/ocr2keeper/evm21/feed_lookup.go:420]: good notes 👍

File: core/services/ocr2/plugins/ocr2keeper/evm21/feed_lookup.go [hunk 2]
```diff
@@ -23,21 +23,25 @@ import (
...
+	applicationJson     = "application/json"
+	blockNumber         = "blockNumber" // valid for v0.2
+	feedID              = "feedID"      // valid for v0.3
+	feedIdHex           = "feedIdHex"   // valid for v0.2
+	headerAuthorization = "Authorization"
+	headerContentType   = "Content-Type"
+	headerTimestamp     = "X-Authorization-Timestamp"
+	headerSignature     = "X-Authorization-Signature-SHA256"
+	headerUpkeepId      = "X-Authorization-Upkeep-Id"
+	mercuryPathV2       = "/client?"
```
[the remark this answers is not in the document]
Comment (FelixFan1992) [reply core/services/ocr2/plugins/ocr2keeper/evm21/feed_lookup.go:40]: done

[the remark this answers is not in the document]
Comment (FelixFan1992) [reply core/services/ocr2/plugins/ocr2keeper/evm21/feed_lookup.go:40]: done

**Review by FelixFan1992 — COMMENTED** [review]
@ad0ll hi TJ, would appreciate your review here. especially mercury response json tags, URL params, etc

**Review by cmalec — APPROVED** [review]

**Pull request merged**